lib/settings.js

```diff
@@ -89,20 +90,27 @@ class Settings extends EventEmitter {
    * @returns {string}
    * @private
    */
-  _getSettingsFilePath() {
+  _getSettingsFilePath() {    
     return this._customSettingsFilePath ?
       this._customSettingsFilePath :
       this._defaultSettingsFilePath;
   }
 
+  _handleObjectChanged() {    
+    const settingsFilePath = this._getSettingsFilePath();        
+    jsonfile.writeFileSync(settingsFilePath, this._cacheObj);    
+    _.forEach(this._watchHandlers, (handler) => {     
+      handler.populate()        
```

Comment:
Instead of this, we should also pass in the keypath, no? Then we can first filter out the keypath that is not included? 

We can also use https://lodash.com/docs/4.17.5#at to generate the paths of the changed object.

Reply:
I think the implementation is not effecient, this means we'd need to call all handlers populate function regardless if the keypath is even part of the set object or not.

Also, the handle function shouldn't be called populate, it should be handle, or onChange?

This should also handle error so handler error doesn't affect other handlers

Reply:
yeah..we just inform the whole object is changed. But in Observer instance we just call handler if the value is changed.

i have no simple/good way to extract the path change of an object. Not sure lodash have any method that we can detect the path change in object??
E.g: We have 2 objects such as  BeforeObject, AfterObject => can we extract all paths that have change?

We have to consider the case the user can set children object. E.g

obj.a.b.c.d

if user set obj.a.b = {....}
=> we have to notify to subscribers that watch on b,c,d ...

Reply:
hm...i think i can do somethink like...

let needNotifiedHandlers = _.filter(this._handlers, (f) => return f.keyPath.startWith('path change')